File: com.github.tux2323.demo.chat.server/src/com/github/tux2323/demo/chatserver/internal/Activator.java
package com.github.tux2323.demo.chatserver.internal;

import org.osgi.framework.BundleActivator;
import org.osgi.framework.BundleContext;

import com.github.tux2323.demo.chatserver.ChatServer;

public class Activator implements BundleActivator {

	private static BundleContext context;

	static BundleContext getContext() {
		return context;
	}
	
	@Override
	public void start(BundleContext bundleContext) throws Exception {
		Activator.context = bundleContext;
		ChatServer chatService = new ChatServiceImpl();
		context.registerService(ChatServer.class.getName(), chatService, null);
	}

	@Override
	public void stop(BundleContext bundleContext) throws Exception {
		Activator.context = null;
	}

}

File: com.github.tux2323.demo.chat.server/src/com/github/tux2323/demo/chatserver/internal/Activator.scala
package com.github.tux2323.demo.chatserver.internal

import com.github.tux2323.demo.chatserver._

import org.osgi.framework.BundleActivator;
import org.osgi.framework.BundleContext;

class Activator extends BundleActivator {

	
	
	override def start(context : BundleContext) {
		ApplicationContext.context = context;
		var chatServer = new ChatServerImpl();
		context.registerService(classOf[ChatServer].getName(), chatServer, null);
	}
	
	override def stop(context : BundleContext) {
		ApplicationContext.context = null;
	}
	
	
	
}